mitosheet/src/components/taskpanes/SnowflakeImport/SnowflakeImportTaskpane.tsx

```diff
@@ -0,0 +1,478 @@
+import React, { useEffect, useState } from "react";
+import useSendEditOnClick from '../../../hooks/useSendEditOnClick';
+import MitoAPI from "../../../jupyter/api";
+import { AnalysisData, SheetData, StepType, UIState, UserProfile } from "../../../types";
+import { toggleInArray } from "../../../utils/arrays";
+
+import { classNames } from "../../../utils/classNames";
+import DropdownItem from "../../elements/DropdownItem";
+import Input from "../../elements/Input";
+import MultiToggleBox from "../../elements/MultiToggleBox";
+import MultiToggleItem from "../../elements/MultiToggleItem";
+import Select from "../../elements/Select";
+import TextButton from "../../elements/TextButton";
+import Col from "../../layout/Col";
+import CollapsibleSection from "../../layout/CollapsibleSection";
+import Row from "../../layout/Row";
+import Spacer from "../../layout/Spacer";
+import DefaultEmptyTaskpane from "../DefaultTaskpane/DefaultEmptyTaskpane";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+
+
+interface SnowflakeImportTaskpaneProps {
+    mitoAPI: MitoAPI;
+    userProfile: UserProfile;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+}
+
+export type SnowflakeCredentialsValidityCheckResult = {type: 'success'} | {type: 'error', 'error_message': string}
+
+export type SnowflakeCredentials = {type: 'username/password', username: string, password: string, account: string};
+
+export type SnowflakeConnection = {
+    warehouse: string | undefined, 
+    database: string | undefined, 
+    schema: string | undefined,
+    table: string | undefined
+}
+
+export type SnowflakeConnectionAPIReturn = {
+    warehouse: string | null, 
+    database: string | null, 
+    schema: string | null,
+    table: string | null
+}
```

Comment:
Gross! Why don't we just make `SnowflakeConnection` (which we should rename) have `warehouse: string | undefined | null`?